sonar-plugin-api/src/main/java/org/sonar/api/batch/rule/Checks.java

```diff
@@ -110,10 +110,18 @@ private void add(RuleKey ruleKey, C obj) {
   }
 
   public Checks<C> addAnnotatedChecks(Object... checkClassesOrObjects) {
-    return addAnnotatedChecks(Arrays.asList(checkClassesOrObjects));
+    return addAnnotatedChecks((Iterable) Arrays.asList(checkClassesOrObjects));
   }
 
+  /**
+   * @deprecated since 5.2 use {@link #addAnnotatedChecks(Iterable)}
+   */
+  @Deprecated
   public Checks<C> addAnnotatedChecks(Collection checkClassesOrObjects) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)